### PR TITLE
Added type names for exposed metatables

### LINKED ISSUE #168: Expose type of userdata and tables through metatables __type
Hi, 

I wonder whether you would accept exposing tzpe of userdatas and tables through `__type` in metatables. It is easier to determine real userdata/table type this way than using tostring. Of course I can add it and make pull requested. For now just want to know whether you would accept such change.

*I work right now into integrating lua-cURLv3 more closely into [eli@0.13.0](https://github.com/cryon-io/eli/tree/0.13.0) and providing simple RestClient implementation based on it. Raw lua-cURLv3 is available for other scenarios that this client won't cover. I can inject `__type` into all your metatables on my side of course, but it feels more reasonable to do it on your side and making it generally available feature.*

### PR DESCRIPTION
**Pull request by cryi**
Fixes #168.

### CODE REVIEW

**Comment by cryi** [comment]
Not ready to be merged.

**Comment by moteus** [comment]
Sorry for delay
`LcURL` is a prefix for C data types.
And I think this field should be added to metadata in a C code as well

**Comment by cryi** [comment]
Hi @moteus,

the LcURL prefix should be there. All metatables in C have their name defined with that prefix. For example here <https://github.com/Lua-cURL/Lua-cURLv3/blob/833e87c830bed05fe3910a33f573c202a48ba6d4/src/lceasy.c#L24>
That name is passed to your utility method for [creating metatables](https://github.com/Lua-cURL/Lua-cURLv3/blob/3f119d585c7c6aba520f02fdf0d9bc322fe5547b/src/l52util.c#L63-L79) and name is injected as `__type`.

So I believe I already did what you are asking 🤔

**Comment by cryi** [comment]
@moteus ping.

File: src/lua/cURL/impl/cURL.lua [hunk 1]
```diff
@@ -717,7 +718,7 @@ end
 end
 -------------------------------------------
 
-setmetatable(cURL, {__index = curl})
+setmetatable(cURL, {__index = curl, __type = "LCURL"})
```
Comment (moteus) [inline src/lua/cURL/impl/cURL.lua:721]: This is not a class but just module. I do not sure that it should have a type metamethod

**Comment by cryi** [comment]
@moteus fixed.

**Pull request merged**